### PR TITLE
prefer to using using system protoc and antlr4 in mill

### PR DESCRIPTION
**Pull request by sequencer**
This PR make FIRRTL be able to compile fully offline with correct ivy caches.
It is able to use system protoc and antlr4 now.

### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you update the FIRRTL spec to include every new feature/behavior?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement
- new feature/API

#### API Impact
None

#### Backend Code Generation Impact
None

#### Desired Merge Strategy
- Rebase: You will rebase the PR onto master and it will be merged with a merge commit.

#### Release Notes
None

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (1.2.x, 1.3.0, 1.4.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Review by jackkoenig — DISMISSED** [review]
What if the system one is the wrong version? Too old? Too new?

**Comment by sequencer** [comment]
Sounds reasonable. I will check version in the patten match.

File: build.sc [hunk 1]
```diff
@@ -123,55 +159,80 @@ class firrtlCrossModule(val crossScalaVersion: String) extends CrossSbtModule wi
...
+          T.ctx.dest / unpackPath / "bin" / "protoc.exe"
+        else
+          T.ctx.dest / unpackPath / "bin" / "protoc"
+
+        if (!os.exists(bin))
+          Util.downloadUnpackZip(
+            s"https://github.com/protocolbuffers/protobuf/releases/download/v$protocVersion/protoc-$protocVersion-$protocBinary.zip",
+            unpackPath
+          )
+        // Download Linux/Mac binary doesn't have x.
```
Comment (jackkoenig) [inline build.sc:227]:
```suggestion
        if (!os.exists(bin)) {
          Util.downloadUnpackZip(
            s"https://github.com/protocolbuffers/protobuf/releases/download/v$protocVersion/protoc-$protocVersion-$protocBinary.zip",
            unpackPath
          )
        }
        // Download Linux/Mac binary doesn't have x.
```

**Comment by sequencer** [comment]
Asking for review from @jackkoenig again.

**Review by seldridge — APPROVED** [review]

**Pull request merged**